apps/total-typescript/src/team/index.tsx

```diff
@@ -23,28 +24,41 @@ const InviteTeam: React.FC<React.PropsWithChildren<InviteTeamProps>> = ({
   session,
   setPersonalPurchase,
 }) => {
-  const redemptionsLeft =
-    purchase.bulkCoupon &&
-    purchase.bulkCoupon.maxUses > purchase.bulkCoupon.usedCount
-  const numberOfRedemptionsLeft =
-    purchase?.bulkCoupon &&
-    purchase?.bulkCoupon.maxUses - purchase.bulkCoupon.usedCount
+  const bulkCouponSchema = z
+    .object({maxUses: z.number(), usedCount: z.number()})
+    .transform(({maxUses, usedCount}) => {
+      return {
+        maxUses,
+        usedCount,
+        numberOfRedemptionsLeft: maxUses - usedCount,
+        redemptionsLeft: maxUses > usedCount,
+      }
+    })
+
+  const bulkCouponData = bulkCouponSchema.parse(purchase.bulkCoupon)
```

Comment:
I feel good about this refactoring -- it cleans up the type guard / presence checking by using a declarative Zod transformation that exposes the original two values plus the two computed values.

Reply:
looks great, `redemptionsLeft` is phrased like a number not a boolean and could use a `has` prefix imo

Reply:
good catch -- the phrasing felt off, I like the sound of `hasRedemptionsLeft`